cmd/virtual-kubelet/internal/commands/root/root.go

```diff
@@ -228,33 +225,39 @@ func runRootCommand(ctx context.Context, s *provider.Store, c Opts) error {
 		}
 	}()
 
+	setNodeReady(pNode)
+	if err := np.UpdateStatus(ctx, pNode); err != nil {
+		return errors.Wrap(err, "error marking the node as ready")
+	}
 	log.G(ctx).Info("Initialized")
 
 	<-ctx.Done()
 	return nil
 }
 
-func newClient(configPath string) (*kubernetes.Clientset, error) {
-	var config *rest.Config
-
-	// Check if the kubeConfig file exists.
-	if _, err := os.Stat(configPath); !os.IsNotExist(err) {
-		// Get the kubeconfig from the filepath.
-		config, err = clientcmd.BuildConfigFromFlags("", configPath)
-		if err != nil {
-			return nil, errors.Wrap(err, "error building client config")
+func setNodeReady(n *corev1.Node) {
+	for i, c := range n.Status.Conditions {
+		if c.Type != "Ready" {
+			continue
 		}
-	} else {
-		// Set to in-cluster config.
-		config, err = rest.InClusterConfig()
-		if err != nil {
-			return nil, errors.Wrap(err, "error building in cluster config")
-		}
-	}
 
-	if masterURI := os.Getenv("MASTER_URI"); masterURI != "" {
-		config.Host = masterURI
+		c.Message = "Kubelet is ready"
+		c.Reason = "KubeletReady"
+		c.Status = corev1.ConditionTrue
+		c.LastHeartbeatTime = metav1.Now()
+		c.LastTransitionTime = metav1.Now()
+		n.Status.Conditions[i] = c
+		return
 	}
 
-	return kubernetes.NewForConfig(config)
+	// No ready condition in node status
```

Comment:
We should add the default diskpressure, memorypressure, and (I think pid pressure) to this.

Reply:
I'd rather just take this out. If it fails its a provider problem (and specifically this is for e2e in this repo only anyway).

Reply:
Updated, took this special condition out since it's not going to be reached anyway.